src/averageAges.js

```diff
@@ -15,11 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMales = people.filter(({ sex, died }) => {
+    if (century) {
+      return sex === 'm' && Math.ceil(died / 100) === century;
+    }
+
+    return sex === 'm';
+  });
+
+  const summaryAge = onlyMales.reduce(
+    (acc, male) => acc + (male.died - male.born),
+    0
+  );
+
+  const averageAge = +(summaryAge / onlyMales.length).toFixed(2);
```

Comment:
move it to the separate function, and use this function here and in the `calculateWomenAverageAge`